### ISSUE TITLE
Inaccurate type of `setErrorHandler`

### ISSUE THREAD

**Issue by chengluyu** [fastify/fastify#2091]
## 🐛 Bug Report

Let's see the definition of `setErrorHandler`.

https://github.com/fastify/fastify/blob/14b4e02d3648355987568eccf328a7111558c1b2/fastify.d.ts#L696

As you see, the type of the first parameter is `FastifyError`, which extends `Error` and has two optional properties.

https://github.com/fastify/fastify/blob/14b4e02d3648355987568eccf328a7111558c1b2/fastify.d.ts#L52-L58

However, the parameter `error` may be any type, because this method is intended to set a handler that deals with various uncaught errors from routes, plugins, etc. You can even throw some customized objects in routes and handle them in the handler.

## To Reproduce

```ts
import Fastify from 'fastify';

const fastify = Fastify();

fastify.setErrorHandler(async (error, request, reply) => {
  // 💥 Property 'hello' does not exist on type 'FastifyError'.ts(2339)
  reply.send(error.hello);
});

fastify.route({
  method: 'GET',
  url: "/",
  async handler(request, reply) {
    // Raise some customized error object
    throw { hello: 'world' };
  }
});
```

## Expected behavior

Well, there should be no error.

## Your Environment

fastify@2.12.0

**Comment by mcollina** [comment]
Would you like to send a PR to fix?

**Comment by chengluyu** [comment]
> Would you like to send a PR to fix?

Hi, I've sent the PR. Please let me know if there is anything I could do next.

**Comment by chengluyu** [comment]
Closed because of #2092
